src/panels/SkyBrowserPanel/hooks.ts

```diff
@@ -0,0 +1,223 @@
+import { useEffect } from 'react';
+
+import { useOpenSpaceApi } from '@/api/hooks';
+import { useAppDispatch, useAppSelector } from '@/redux/hooks';
+import {
+  subscribeToSkyBrowser,
+  unsubscribeToSkyBrowser
+} from '@/redux/skybrowser/skybrowserMiddleware';
+import {
+  setActiveImage,
+  setImageCollectionData
+} from '@/redux/skybrowser/skybrowserSlice';
+
+export function useGetWwtImageCollection() {
+  const luaApi = useOpenSpaceApi();
+  const dispatch = useAppDispatch();
+  useEffect(() => {
+    async function getWwtListOfImages() {
+      if (!luaApi || !luaApi.skybrowser) {
+        return;
+      }
+      const imgData = await luaApi.skybrowser.listOfImages();
+      if (imgData) {
+        const images = Object.values(imgData);
+        if (images.length !== 0) {
+          const imgDataWithKey = images.map((image) => ({
+            ...image,
+            key: image.identifier
+          }));
+          dispatch(setImageCollectionData(imgDataWithKey));
+        }
+      } else {
+        throw new Error('No AAS WorldWide Telescope images!');
+      }
+    }
+    try {
+      getWwtListOfImages();
+    } catch (e) {
+      throw Error(`Could not load image collection from OpenSpace: ${e}`);
+    }
+  }, [luaApi, dispatch]);
+}
+
+export function useGetSkyBrowserData() {
+  useGetWwtImageCollection();
+  const dispatch = useAppDispatch();
+  useEffect(() => {
+    dispatch(subscribeToSkyBrowser());
+    return () => {
+      dispatch(unsubscribeToSkyBrowser());
+    };
+  }, [dispatch]);
+}
+
+export function useActiveImage(): [string, (url: string) => void] {
+  const activeImage = useAppSelector((state) => state.skybrowser.activeImage);
+
+  const dispatch = useAppDispatch();
+
+  function setImage(url: string) {
+    dispatch(setActiveImage(url));
+  }
+  return [activeImage, setImage];
+}
+
+// Below are hooks to get the values from the sky browser topic. They
+// are setup so the value from the selector only should update when
+// the value itself actually updates. TODO (ylvse 2025-01-29): rewrite
+// the topic so that most of the data are properties instead, and only
+// pass the data for the currently selected browser
+
+export function useSelectedBrowserColor(): number[] | undefined {
+  const color = useAppSelector(
+    (state) => state.skybrowser.selectedBrowser?.color,
+    lowPrecisionEqualArray
+  );
+  return color;
+}
+
+export function useSelectedBrowserColorString(): string | undefined {
+  const color = useSelectedBrowserColor();
+  return color ? (`rgb(${color.join(',')})` as string) : undefined;
+}
+
+export function useSelectedBrowserRadius(): number | undefined {
+  const radius = useAppSelector(
+    (state) => state.skybrowser.selectedBrowser?.borderRadius,
+    lowPrecisionEqual()
+  );
+  return radius;
+}
+
+export function lowPrecisionEqualMatrix(
+  lhs: number[][] | undefined,
+  rhs: number[][] | undefined
+) {
+  if (lhs === undefined && rhs === undefined) {
```

Comment:
All these three functions use the same check, which is a bit nasty/fragile to me.  Could the functions be generalized somehow?

Reply:
Maybe add a function that does this very check?